### ISSUE TITLE
XCP-ng not sending hostname back via DHCP

### ISSUE THREAD

**Issue by biztems** [xcp-ng/xcp#98]
Hi guys,

First things first: thanks for the magnificent work on xcp-ng. We've successfully migrated all of our machines from XS 7.2 to XCP-NG 7.5/7.6 with no issues at all. No small feat!

Anyway, this is something that was happening back on stock XS, from what I remember... The Dom0 isn't sending back any hostname via DHCP. It's a bit of an issue for us since we use hostnames for pretty much everything as it's so much easier than digging into DHCP leases every time.

Also, this applies to XCP-ng Control Center too, since we can't use hostnames when adding hosts to a pool. Ugh!

Apparently, from what we can see, it has something to do with the fact that there's no `send host-name` specified in `/var/lib/xcp/dhclient-xenbr0.conf`. I've tried adding it manually, but xcp-ng keeps removing it (kinda expected).

Is there some hack we can do to enable this functionaly back?
Stock CentOS works fine and sends back the hostname as expected.

Thanks for your time.
Best Regards

**Comment by olivierlambert** [comment]
This is interesting, we'll take a look and see where this is modified (and why). If you have any idea, extra-pointers are welcome :+1:

**Comment by webdawg** [comment]
I can confirm the same results here.

**Comment by olivierlambert** [comment]
Is it the same behavior in XS 7.6?

**Comment by webdawg** [comment]
I do not know, I installed xcp-ng yesterday, and I am getting rid of all my xenserver boxes.  I stopped upgrading xenserver when they started to pull features.

**Comment by olivierlambert** [comment]
The thing is we need to understand if it's something we introduce or if it's a XS problem that we could fix :) Try in a nested VM if you can, and report the result, this way it will be easier to solve this!

**Comment by biztems** [comment]
> The thing is we need to understand if it's something we introduce or if it's a XS problem that we could fix :) Try in a nested VM if you can, and report the result, this way it will be easier to solve this!

Finally found some time to test this - same behaviour in XS 7.6, when ran in a VM. It's definitely from upstream then.

**Comment by stormi** [comment]
Then could you report it to bugs.xenserver.org and add the link here so that we can follow how they answer to it?

**Comment by biztems** [comment]
> Then could you report it to bugs.xenserver.org and add the link here so that we can follow how they answer to it?

Sure, here it is: https://bugs.xenserver.org/browse/XSO-933

**Comment by stormi** [comment]
Thanks!

**Comment by stormi** [comment]
Update candidates are available that fix this issue, on XCP-ng 8.0 and 8.1.

Run: 
```
yum clean all --enablerepo=xcp-ng-testing
yum update xcp-networkd --enablerepo=xcp-ng-testing
xe-toolstack-restart
```

**Comment by stormi** [comment]
Since there are two issues open for this bug, I'm closing one. See #363.
